INFO - PPO - Running command 'bc_run'
INFO - PPO - Started run with ID "70"
Creating env with params {'RUN_TYPE': 'ppo3', 'SEEDS': [9456, 1887, 5578, 5987, 516], 'LOCAL_TESTING': False, 'EX_NAME': 'testing_lta', 'SAVE_DIR': 'testing_lta/', 'GPU_ID': 0, 'PPO_RUN_TOT_TIMESTEPS': 8000000.0, 'mdp_params': {'layout_name': 'simple', 'start_order_list': None, 'rew_shaping_params': {'ONION_IN_EMPTY_POT_REWARD': 3, 'ONION_IN_PARTIAL_POT_REWARD': 3, 'DISH_PICKUP_REWARD': 2, 'SOUP_PICKUP_FROM_READY_POT_REWARD': 2, 'BOTH_POTS_FULL_REWARD': 5, 'SERVE_SOUP_REWARD': 2}}, 'env_params': {'horizon': 400}, 'mdp_generation_params': {'padded_mdp_shape': [11, 7], 'mdp_shape_fn': [[5, 11], [5, 7]], 'prop_empty_fn': [0.6, 1], 'prop_feats_fn': [0, 0.6]}, 'ENTROPY': 0.1, 'GAMMA': 0.99, 'sim_threads': 30, 'TOTAL_BATCH_SIZE': 12000, 'BATCH_SIZE': 400, 'MAX_GRAD_NORM': 0.1, 'LR': 0.001, 'LR_ANNEALING': 3, 'VF_COEF': 0.5, 'STEPS_PER_UPDATE': 8, 'MINIBATCHES': 10, 'CLIPPING': 0.05, 'LAM': 0.98, 'SELF_PLAY_HORIZON': [500000.0, 3000000.0], 'REW_SHAPING_HORIZON': 1000000.0, 'OTHER_AGENT_TYPE': 'bc_train', 'HM_PARAMS': [True, 0.3], 'NUM_HIDDEN_LAYERS': 3, 'SIZE_HIDDEN_LAYERS': 64, 'NUM_FILTERS': 25, 'NUM_CONV_LAYERS': 3, 'NETWORK_TYPE': 'two_head_conv_and_mlp2', 'SAVE_BEST_THRESH': 50, 'TRAJECTORY_SELF_PLAY': True, 'VIZ_FREQUENCY': 50, 'grad_updates_per_agent': 53280.0}
Computing MediumLevelPlanner to be saved in /home/mzhao2/long_term_adaptation/human_aware_rl/overcooked_ai/overcooked_ai_py/data/planners/simple_am.pkl
It took 0.1429603099822998 seconds to create mlp
num_convs 3
2022-02-16 20:50:45.095782: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2022-02-16 20:50:45.105178: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2999645000 Hz
2022-02-16 20:50:45.109238: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x561367bcf440 executing computations on platform Host. Devices:
2022-02-16 20:50:45.109282: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
WARNING:tensorflow:From /home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING - tensorflow - From /home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
SHAPE OF X (30, 5, 4, 20)
WARNING:tensorflow:From ../../human_aware_rl/baselines_utils_for_lta.py:237: conv2d (from tensorflow.python.layers.convolutional) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.conv2d instead.
WARNING - tensorflow - From ../../human_aware_rl/baselines_utils_for_lta.py:237: conv2d (from tensorflow.python.layers.convolutional) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.conv2d instead.
WARNING:tensorflow:From /home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING - tensorflow - From /home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From ../../human_aware_rl/baselines_utils_for_lta.py:252: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING - tensorflow - From ../../human_aware_rl/baselines_utils_for_lta.py:252: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From ../../human_aware_rl/baselines_utils_for_lta.py:255: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.dense instead.
WARNING - tensorflow - From ../../human_aware_rl/baselines_utils_for_lta.py:255: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.dense instead.
Last layer conv network output shape (30, 64)
FINAL OUTPUT concat network output shape (30, 12)
123801
number of parameters None
SHAPE OF X (1200, 5, 4, 20)
Last layer conv network output shape (1200, 64)
FINAL OUTPUT concat network output shape (1200, 12)
123916
number of parameters None
> /home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/ppo3/model.py(101)__init__()
-> human_model_prediction = copy.deep_copy(train_model.q).eval()[:, 6:]
(Pdb) *** AttributeError: module 'copy' has no attribute 'deep_copy'
(Pdb) *** TypeError: can't pickle _thread.RLock objects
(Pdb) <tf.Tensor 'strided_slice:0' shape=(1200, 6) dtype=float32>
(Pdb) *** tensorflow.python.framework.errors_impl.FailedPreconditionError: Attempting to use uninitialized value ppo_agent/ppo2_model/pi/robot_conv_initial/kernel
	 [[node ppo_agent/ppo2_model/pi/robot_conv_initial/kernel/read (defined at ../../human_aware_rl/baselines_utils_for_lta.py:237) ]]

Caused by op 'ppo_agent/ppo2_model/pi/robot_conv_initial/kernel/read', defined at:
  File "../lta_ppo/train_model_on_bc_only.py", line 247, in <module>
    @ex.automain
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/sacred-0.7.4-py3.7.egg/sacred/experiment.py", line 137, in automain
    self.run_commandline()
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/sacred-0.7.4-py3.7.egg/sacred/experiment.py", line 260, in run_commandline
    return self.run(cmd_name, config_updates, named_configs, {}, args)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/sacred-0.7.4-py3.7.egg/sacred/experiment.py", line 209, in run
    run()
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/sacred-0.7.4-py3.7.egg/sacred/run.py", line 221, in __call__
    self.result = self.main_function(*args)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/sacred-0.7.4-py3.7.egg/sacred/config/captured_function.py", line 46, in captured_function
    result = wrapped(*args, **kwargs)
  File "../lta_ppo/train_model_on_bc_only.py", line 286, in bc_run
    model = create_model(gym_env, "ppo_agent", **params)
  File "../../human_aware_rl/baselines_utils_for_lta.py", line 479, in create_model
    network_kwargs=kwargs
  File "/home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/ppo3/ppo3.py", line 126, in learn_human_model
    max_grad_norm=max_grad_norm, scope=scope)
  File "/home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/ppo3/model.py", line 41, in __init__
    act_model = policy(nbatch_act, 1, sess)
  File "/home/mzhao2/long_term_adaptation/human_aware_rl/baselines/baselines/common/policies.py", line 221, in policy_fn
    policy_latent = policy_network(encoded_x)
  File "../../human_aware_rl/baselines_utils_for_lta.py", line 237, in network_fn
    name="robot_conv_initial"
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/util/deprecation.py", line 324, in new_func
    return func(*args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/layers/convolutional.py", line 424, in conv2d
    return layer.apply(inputs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1227, in apply
    return self.__call__(inputs, *args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/layers/base.py", line 530, in __call__
    outputs = super(Layer, self).__call__(inputs, *args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py", line 538, in __call__
    self._maybe_build(inputs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1603, in _maybe_build
    self.build(input_shapes)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/keras/layers/convolutional.py", line 165, in build
    dtype=self.dtype)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/layers/base.py", line 435, in add_weight
    getter=vs.get_variable)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py", line 349, in add_weight
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/training/checkpointable/base.py", line 607, in _add_variable_with_custom_getter
    **kwargs_for_getter)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 1479, in get_variable
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 1220, in get_variable
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 547, in get_variable
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 499, in _true_getter
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 911, in _get_single_variable
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 213, in __call__
    return cls._variable_v1_call(*args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 176, in _variable_v1_call
    aggregation=aggregation)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 155, in <lambda>
    previous_getter = lambda **kwargs: default_variable_creator(None, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variable_scope.py", line 2495, in default_variable_creator
    expected_shape=expected_shape, import_scope=import_scope)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 217, in __call__
    return super(VariableMetaclass, cls).__call__(*args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 1395, in __init__
    constraint=constraint)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/variables.py", line 1557, in _init_from_args
    self._snapshot = array_ops.identity(self._variable, name="read")
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py", line 81, in identity
    ret = gen_array_ops.identity(input, name=name)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 3890, in identity
    "Identity", input=input, name=name)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/framework/ops.py", line 3300, in create_op
    op_def=op_def)
  File "/home/mzhao2/anaconda3/envs/lta_overcooked/lib/python3.7/site-packages/tensorflow/python/framework/ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()

FailedPreconditionError (see above for traceback): Attempting to use uninitialized value ppo_agent/ppo2_model/pi/robot_conv_initial/kernel
	 [[node ppo_agent/ppo2_model/pi/robot_conv_initial/kernel/read (defined at ../../human_aware_rl/baselines_utils_for_lta.py:237) ]]
(Pdb) (Pdb) <tf.Tensor 'Sqrt:0' shape=() dtype=float32>
(Pdb) *** TypeError: subtract() got an unexpected keyword argument 'axis'
(Pdb) *** TypeError: subtract() got an unexpected keyword argument 'axis'
(Pdb) *** SyntaxError: invalid syntax
(Pdb) *** AttributeError: module 'tensorflow' has no attribute 'sub'
(Pdb) <tf.Tensor 'sub_11:0' shape=(1200, 6) dtype=float32>
(Pdb) <tf.Tensor 'Square_4:0' shape=(1200, 6) dtype=float32>
(Pdb) <tf.Tensor 'Sum_3:0' shape=(1200,) dtype=float32>
(Pdb) 